### PR TITLE
MODBULKOPS-32: FQM Query Tool Metadata supporting. Initial commit

### PR DESCRIPTION
**Pull request by DmytroBykov1**
## Purpose
https://issues.folio.org/browse/MODBULKOPS-32
FQM Query Tool Metadata supporting
 

## Approach
1. Add https://github.com/folio-org/folio-query-tool-metadata as Maven dependency.
2. Prepare API skeleton - implement the EntityTypesApi and FqlQueryApi interfaces as EntityTypesController and FqlQueryController.
 

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/bulkops/controller/EntityTypesController.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.folio.bulkops.controller;
+
+import org.folio.querytool.rest.resource.EntityTypesApi;
+
+public class EntityTypesController implements EntityTypesApi {
```
Comment (khandramai) [inline src/main/java/org/folio/bulkops/controller/EntityTypesController.java:5]: Please, create stubs forEntityTypesApi interface.

File: src/main/java/org/folio/bulkops/controller/FqlQueryController.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package org.folio.bulkops.controller;
+
+import org.folio.querytool.rest.resource.FqlQueryApi;
+
+public class FqlQueryController implements FqlQueryApi {
```
Comment (khandramai) [inline src/main/java/org/folio/bulkops/controller/FqlQueryController.java:5]: Please, create stubs for FqlQueryApi interface.

File: pom.xml [hunk 3]
```diff
@@ -35,6 +35,7 @@
     <folio-spring-base.version>7.2.0</folio-spring-base.version>
     <folio-spring-cql.version>7.2.0</folio-spring-cql.version>
     <folio-service-tools.version>3.1.0</folio-service-tools.version>
+    <folio-query-tool-metadata.version>1.1.0-SNAPSHOT</folio-query-tool-metadata.version>
```
Comment (khandramai) [inline pom.xml:38]: Please, change 1.1.0-SNAPSHOT version to released one https://github.com/folio-org/folio-query-tool-metadata/releases/tag/v1.0.0.

File: src/main/java/org/folio/bulkops/controller/EntityTypesController.java (hunk 1 above)
Comment (DmytroBykov1) [reply src/main/java/org/folio/bulkops/controller/EntityTypesController.java:15]: done

File: src/main/java/org/folio/bulkops/controller/FqlQueryController.java (hunk 2 above)
Comment (DmytroBykov1) [reply src/main/java/org/folio/bulkops/controller/FqlQueryController.java]: done

File: pom.xml (hunk 3 above)
Comment (DmytroBykov1) [reply pom.xml]: done

**Comment by NikitaSedyx** [comment]
I believe you override endpoints from FQL, in okapi can't live 2 endpoints with the same name - except cases when module header is passed

**Comment by khandramai** [comment]
NOTE: separate yaml-file is needed?

**Comment by NikitaSedyx** [comment]
I believe not only yaml file, module descriptor should be updated to describe permissions and API

**Comment by khandramai** [comment]
I assume we need to consider multiple interfaces https://github.com/folio-org/okapi/blob/master/doc/guide.md#multiple-interfaces. @TarasSpashchenko, please, confirm. If yes, @DmytroBykov1 , please, follow the instruction above.

**Review by siarhei-charniak — APPROVED** [review]

**Pull request closed without merging**